client/adventureMap/CList.cpp

```diff
@@ -301,6 +305,72 @@ void CHeroList::CHeroItem::gesture(bool on, const Point & initialPosition, const
 	GH.windows().createAndPushWindow<RadialMenu>(pos.center(), menuElements, true);
 }
 
+bool CHeroList::CHeroItem::captureThisKey(EShortcut key)
+{
+
+	switch(key)
+	{
+	case EShortcut::LIST_HERO_UP:
+	case EShortcut::LIST_HERO_DOWN:
+	case EShortcut::LIST_HERO_TOP:
+	case EShortcut::LIST_HERO_BOTTOM:
+	case EShortcut::LIST_HERO_DELETE:
+		return hero && pos.isInside(GH.getCursorPosition());
+	}
+	return false;
+}
+
+void CHeroList::CHeroItem::keyPressed(EShortcut key)
+{
+	if(!hero)
+		return;
+
+	if(!pos.isInside(GH.getCursorPosition()))
+		return;
+
+	auto & heroes = LOCPLINT->localState->getWanderingHeroes();
+
+	if(key == EShortcut::LIST_HERO_DELETE)
+	{
+		CFunctionList<void()> ony = [=](){ };
```

Comment:
What does ony mean?

Reply:
I think `on yes`. Also used in another place in VCMI code base like this. Removed. As it doesn't make sense to store lambda in var.